### ISSUE TITLE
useBackgroundBufferMutex = 0 is default

### ISSUE THREAD

**Issue by mhantke** [antonbarty/cheetah-2017#10]
Why do we use useBackgroundBufferMutex = 0 as a default? Isn't that risky? This was probably done in the sake of backwards compatibility. I suggest to change it to 1.

**Comment by cnettel** [comment]
It’s a bit risky. It is also slow to turn it on. I seem to remember that the mutex is held for quite a while. One should at least do a benchmark to see how severe the effect is with heavy threading.

/Carl

Från: Max Hantke [mailto:notifications@github.com]
Skickat: den 10 oktober 2014 13:34
Till: antonbarty/cheetah
Ämne: [cheetah] useBackgroundBufferMutex = 0 is default (#10)

Why do we use useBackgroundBufferMutex = 0 as a default? Isn't that risky? This was probably done in the sake of backwards compatibility. I suggest to change it to 1.

**Comment by antonbarty** [comment]
It is risky for a perfect calculation.

It also slows things down.  And since the running background is an average over the last few events presumably it doesn’t matter much from one pixel to the next whether the average/median is calculated over events  [n-1:n-51] or [n:n-50].  So in the interests of speed I often get by with it turned off…

But you are right - strictly speaking this could result in things getting out of sync, for example with different parts of the background calculated over different sets of images.  Whether it really matters in this case is another question. 

A

**Comment by mhantke** [comment]
I agree with you that we should keep the option to increase performance if
needed. But I would like to set the default value to controlled conditions.
Especially for the naive user would be more intuitive.
/Max
